SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testQueryPhaseIsExecutedInAnAvailableNodeWhenAllShardsCanBeSkipped failure

<!--
Please fill out the following information, and ensure you have attempted
to reproduce locally
-->

**Build scan**:
https://gradle-enterprise.elastic.co/s/eyuntluqkrc6u/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests/testQueryPhaseIsExecutedInAnAvailableNodeWhenAllShardsCanBeSkipped#1

**Repro line**:
```
./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' \
  --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testQueryPhaseIsExecutedInAnAvailableNodeWhenAllShardsCanBeSkipped" \
  -Dtests.seed=119FBF889C9916E -Dtests.security.manager=true -Dtests.locale=en \
  -Dtests.timezone=Canada/Newfoundland -Druntime.java=11 -Dtests.fips.enabled=true
```
**Reproduces locally?**:
nope

**Applicable branches**:
[seems](https://build-stats.elastic.co/goto/31ac343cdaf215db4714dd933cdc570e) to be just 7.12, though a PR against master caught it too?

**Failure history**:
https://build-stats.elastic.co/goto/31ac343cdaf215db4714dd933cdc570e

**Failure excerpt**:
```
org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests > testQueryPhaseIsExecutedInAnAvailableNodeWhenAllShardsCanBeSkipped FAILED
    java.lang.AssertionError: 
    Expected: <3>
         but: was <4>
        at __randomizedtesting.SeedInfo.seed([119FBF889C9916E:6E35F52890D149F7]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testQueryPhaseIsExecutedInAnAvailableNodeWhenAllShardsCanBeSkipped(SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java:297)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testQueryPhaseIsExecutedInAnAvailableNodeWhenAllShardsCanBeSkipped" -Dtests.seed=119FBF889C9916E -Dtests.security.manager=true -Dtests.locale=en -Dtests.timezone=Canada/Newfoundland -Druntime.java=11 -Dtests.fips.enabled=true
```

--- a/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java
+++ b/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java
@@ -256,7 +256,7 @@ public class SearchableSnapshotsCanMatchOnCoordinatorIntegTests extends BaseSear
 
         // Block the repository for the node holding the searchable snapshot shards
         // to delay its restore
-        blockDataNode(repositoryName, dataNodeHoldingSearchableSnapshot);
+        blockNodeOnAnyFiles(repositoryName, dataNodeHoldingSearchableSnapshot);
 
         // Force the searchable snapshot to be allocated in a particular node
         Settings restoredIndexSettings = Settings.builder()

